py/selenium/webdriver/common/bidi/browser.py

```diff
@@ -123,16 +123,51 @@ def from_dict(cls, data: dict) -> "ClientWindowInfo":
         Returns:
         -------
             ClientWindowInfo: A new instance of ClientWindowInfo.
+
+        Raises:
+        ------
+            ValueError: If required fields are missing or have invalid types.
         """
-        return cls(
-            client_window=data.get("clientWindow"),
-            state=data.get("state"),
-            width=data.get("width"),
-            height=data.get("height"),
-            x=data.get("x"),
-            y=data.get("y"),
-            active=data.get("active"),
-        )
+        try:
+            client_window = data.get("clientWindow")
+            if not isinstance(client_window, str):
+                raise ValueError("clientWindow must be a string")
+
+            state = data.get("state")
+            if not isinstance(state, str):
+                raise ValueError("state must be a string")
```

Comment:
```suggestion
            state = data["state"]
            if not isinstance(state, str):
                raise ValueError("state must be a string")
            if state not in ClientWindowState.VALID_STATES:
                raise ValueError(f"Invalid state: {state}. Must be one of {ClientWindowState.VALID_STATES}")
```

Since `ClientWindowInfo` is returned from the remote end, we should not access any value using `.get()` and use something like `state = data["state"]` in order to fail early if the required value is not present in the response.

Also, add a check for valid states. You will need to add this in the `ClientWindowInfo` class:
```python
VALID_STATES = {FULLSCREEN, MAXIMIZED, MINIMIZED, NORMAL}
```

Reply:
>  we should not access any value using .get() and use something like state = data["state"] in order to fail early if the required value is not present in the response.

I agree, but note that this will raise `KeyError` if not present. I'm not sure that is the desired exception to raise if it is missing.

Reply:
We are wrapping it inside `try/except` block, so eventually it will raise `ValueError`:
```python
        except (KeyError, TypeError) as e:
            raise ValueError(f"Invalid data format for ClientWindowInfo: {e}")
```